src/algebra/ordered_ring.lean

```diff
@@ -627,6 +637,22 @@ lemma mul_pos_of_neg_of_neg {a b : α} (ha : a < 0) (hb : b < 0) : 0 < a * b :=
 have 0 * b < a * b, from mul_lt_mul_of_neg_right ha hb,
 by rwa zero_mul at this
 
+/-- Pullback an `ordered_ring` under an injective map. -/
+def function.injective.ordered_ring {β : Type*}
+  [has_zero β] [has_one β] [has_add β] [has_mul β] [has_neg β]
+  (f : β → α) (hf : function.injective f) (zero : f 0 = 0) (one : f 1 = 1)
+  (add : ∀ x y, f (x + y) = f x + f y) (mul : ∀ x y, f (x * y) = f x * f y)
+  (neg : ∀ x, f (- x) = - f x) :
+  ordered_ring β :=
+-- I used `erw`, since `rw` did not work.
+{ neg := has_neg.neg,
+  sub := λ a b, a + - b,
+  sub_eq_add_neg := λ a b, by simpa only,
```

Comment:
Your should fill out these fields with function.injective.ring

Reply:
Updated, thanks!  I did not know about the existence of `function.injective.ring`!